### PR TITLE
Enhancement: add max int64 checks

### PR DESCRIPTION
**Pull request by shiqizng**
## Summary

This PR adds max int64 checks to round, application ID and asset ID in API requests.
 
## Test Plan
Add unit tests for the handler and filter methods.

### CODE REVIEW

File: api/handlers.go [hunk 1]
```diff
@@ -159,6 +166,9 @@ func (si *ServerImplementation) verifyHandler(operationID string, ctx echo.Conte
 // LookupAccountByID queries indexer for a given account.
 // (GET /v2/accounts/{account-id})
 func (si *ServerImplementation) LookupAccountByID(ctx echo.Context, accountID string, params generated.LookupAccountByIDParams) error {
+	if params.Round != nil && uint64(*params.Round) > math.MaxInt64 {
+		return badRequest(ctx, errValueExceedingInt64)
+	}
 	if err := si.verifyHandler("LookupAccountByID", ctx); err != nil {
 		return badRequest(ctx, err.Error())
 	}
```
Comment (winder) [inline api/handlers.go:174]: Please make sure all of these happen after `verifyHandler`. If the endpoint is disabled we should make sure that is the returned error

Comment (winder) [reply api/handlers.go:174]: What do you think about returning `404` errors instead of bad request?

File: api/converter_utils.go [hunk 2]
```diff
@@ -588,6 +589,10 @@ func (si *ServerImplementation) transactionParamsToTransactionFilter(params gene
 	filter.ApplicationID = uintOrDefault(params.ApplicationId)
 	filter.Limit = min(uintOrDefaultValue(params.Limit, si.opts.DefaultTransactionsLimit), si.opts.MaxTransactionsLimit)
 	filter.Round = params.Round
+	// Integer > math.Int64
+	if filter.AssetID > math.MaxInt64 || filter.ApplicationID > math.MaxInt64 || (filter.Round != nil && uint64(*filter.Round) > math.MaxInt64) {
```
Comment (Eric-Warehime) [inline api/converter_utils.go:593]: I see that there are a lot more fields in a filter which are `uint64` https://github.com/algorand/indexer/blob/develop/idb/idb.go#L196 

Why are these the only ones we need to do checks for as opposed to checking all of the supplied filters?

File: api/handlers.go (hunk 1 above)
Comment (shiqizng) [reply api/handlers.go:174]: I think 404 makes sense. I'll update.

File: api/converter_utils.go (hunk 2 above)
Comment (shiqizng) [reply api/converter_utils.go]: I moved these checks to validateTransactionFilter and added checks for all uint64 fields except for AssetAmount which is type numeric(20) in the DB table.

File: api/handlers.go [hunk 3]
```diff
@@ -51,6 +52,32 @@ type ServerImplementation struct {
...
+	}
+
+	// algos > math.MaxInt64
+	if (filter.AlgosLT != nil && *filter.AlgosLT > math.MaxInt64) ||
+		(filter.AlgosGT != nil && *filter.AlgosGT > math.MaxInt64) {
+		errorArr = append(errorArr, errValueExceedingInt64)
+	}
+
+	// effectiveAmount > math.MaxInt64
+	if (filter.EffectiveAmountLT != nil && *filter.EffectiveAmountLT > math.MaxInt64) ||
```
Comment (shiqizng) [inline api/handlers.go:76]: Should I update these from *uint64 to *int64 to match the table schema? I think we remove these checks if we update the var types.

File: idb/postgres/postgres.go [hunk 4]
```diff
@@ -582,12 +582,12 @@ func buildTransactionQuery(tf idb.TransactionFilter) (query string, whereArgs []
 		partNumber++
 	}
 	if tf.AssetAmountGT != nil {
-		whereParts = append(whereParts, fmt.Sprintf("(t.txn -> 'txn' -> 'aamt')::bigint > $%d", partNumber))
+		whereParts = append(whereParts, fmt.Sprintf("(t.txn -> 'txn' -> 'aamt')::numeric(20) > $%d", partNumber))
```
Comment (winder) [inline idb/postgres/postgres.go:585]: nice catch, did this cause a failure before?

**Review by winder — APPROVED** [review]

File: idb/postgres/postgres.go (hunk 4 above)
Comment (shiqizng) [reply idb/postgres/postgres.go:585]: it was causing a failure for a test case I just added that has a txn with asset amount maxUint64.

**Review by Eric-Warehime — APPROVED** [review]

**Pull request merged**